workflow/controller/operator.go

```diff
@@ -87,6 +93,13 @@ type wfOperationCtx struct {
 	// preExecutionNodePhases contains the phases of all the nodes before the current operation. Necessary to infer
 	// changes in phase for metric emission
 	preExecutionNodePhases map[string]wfv1.NodePhase
+
+	// wfSpec holds the WorkflowSpec for execution
+	wfSpec *wfv1.WorkflowSpec
+	// entrypoint is the starting point for workflow execution
+	entrypoint string
+	// arguments holds all global argument for workflow.
+	arguments wfv1.Arguments
```

Comment:
Also, it seems that whenever this field is used, only the parameters are accessed. I would change this to
```suggestion
	submissionParameters []wfv1.Parameter
```